R/IDateTime.R

```diff
@@ -12,30 +12,30 @@ as.IDate.default <- function(x, ..., tz = attr(x, "tzone")) {
 
 as.IDate.numeric <- function(x, origin = "1970-01-01", ...) {
   if (origin=="1970-01-01") {
-    # standard epoch
-    setattr(as.integer(x), "class", c("IDate", "Date"))
+    # standard epoch; need copy() since as.integer(integer()) doesn't copy
+    (setattr(copy(as.integer(x)), "class", c("IDate", "Date")))
```

Comment:
If as.integer(x) does convert (normally it will) then it will return a fresh object only to be always copied a 2nd time by this copy().  Just a `class(x)<` should be fine now that R 3.1.0 much improved the copies.